pkg/eks/filecache.go

```diff
@@ -0,0 +1,227 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/gofrs/flock"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	// EksctlGlobalEnableCachingEnvName defines an environment property to enable the cache globally.
+	EksctlGlobalEnableCachingEnvName = "EKSCTL_ENABLE_CREDENTIAL_CACHE"
+	// EksctlCacheFilenameEnvName defines an environment property to configure where the cache file should live.
+	EksctlCacheFilenameEnvName = "EKSCTL_CREDENTIAL_CACHE_FILENAME"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
+// Clock implements Now to return the current time.
+//counterfeiter:generate -o fakes/fake_clock.go . Clock
+type Clock interface {
+	Now() time.Time
+}
+
+// RealClock defines a clock using time.Now()
+type RealClock struct{}
+
+// Now returns the current time.
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+type cachedCredential struct {
+	Credential credentials.Value
+	Expiration time.Time
+}
+
+// FileCacheProvider is a file based AWS Credentials Provider implementing expiry and retrieve.
+type FileCacheProvider struct {
+	credentials      *credentials.Credentials // the underlying implementation that has the *real* Provider
+	cachedCredential cachedCredential         // the cached credential, if it exists
+	profile          string
+	clock            Clock
+}
+
+type cacheFile struct {
+	// a map of profiles to cachedCredentials
+	ProfileMap map[string]cachedCredential `yaml:"profiles"`
+}
+
+// Put puts the given cachedCredential with a given key into the map. It will overwrite
+// if the key already exists.
+func (c *cacheFile) Put(key string, credential cachedCredential) {
+	c.ProfileMap[key] = credential
+}
+
+// Get returns cachedCredential if it exists in the cred store.
+func (c *cacheFile) Get(key string) (credential cachedCredential) {
```

Comment:
```suggestion
func (c *cacheFile) Get(key string) (cachedCredential) {
```

![image](https://user-images.githubusercontent.com/19669095/133123184-f0334991-a30a-4b11-ba30-38c07de29459.png)

Reply:
Pfff :D

Reply:
That's the reason for named return types. :D

Reply:
to irritate people with better taste? :trollface:

Reply:
Pfpfpfpfff, you people are just jealous.